src/databricks/labs/ucx/hive_metastore/pipelines_migrate.py

```diff
@@ -65,11 +68,23 @@ def _populate_pipeline_job_tasks_mapping(self) -> None:
                     self._pipeline_job_tasks_mapping[pipeline_id].append(job_info)
                 logger.info(f"Found job:{job.job_id} task:{task.task_key} associated with pipeline {pipeline_id}")
 
-    def _get_pipelines_to_migrate(self) -> list[PipelineInfo]:
+    def _get_pipelines_to_migrate(self) -> list[str]:
         """
-        Returns the list of pipelines in the current workspace
+        Returns the list of pipelines filtered by the include and exclude list
         """
-        return list(self._pipeline_crawler.snapshot())
+        pipeline_ids_to_migrate = []
+
+        if self._include_pipeline_ids:
+            pipeline_ids_to_migrate = self._include_pipeline_ids
+        else:
+            for pipeline in list(self._pipeline_crawler.snapshot()):
+                pipeline_ids_to_migrate.append(pipeline.pipeline_id)
+
+        if self._exclude_pipeline_ids is not None:
+            for pipeline_id in pipeline_ids_to_migrate:
+                if pipeline_id in self._exclude_pipeline_ids:
+                    pipeline_ids_to_migrate.remove(pipeline_id)
```

Comment:
Typically done using sets

```
```suggestion
        if self._exclude_pipeline_ids is not None:
            pipeline_ids_to_migrate = list(set(pipeline_ids_to_migrate) - set(self._exclude_pipeline_ids))
```

You could change it to a set to begin with and update the signature